### PR TITLE
use com.redhat.devtools.intellij.common.tree.StructureTreeModelFactory (#38)

### LINKED ISSUE #38: use com.redhat.devtools.intellij.common.tree.StructureTreeModelFactory
replace the internal ```StructureTreeModelFactory``` by the version created in ```com.redhat.devtools.intellij.common``` at https://github.com/redhat-developer/intellij-common/pull/7

### PR DESCRIPTION
**Pull request by adietish**
fixes #38 
depends on https://github.com/redhat-developer/intellij-common/pull/7

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -56,5 +57,6 @@ publishPlugin {
 }
 
 repositories {
+    mavenLocal()
```
Comment (jeffmaury) [inline build.gradle:60]: You need to add JBoss Nexus as well if you can CI to succeed

Comment (adietish) [reply build.gradle:60]: @jeffmaury JBoss Nexus is mirrored to/from central, why add a specific source?

Comment (adietish) [reply build.gradle:60]: I seem not to be authorized to publish:
```
> Could not PUT 'https://repository.jboss.org/nexus/content/groups/public/com/redhat/devtools/intellij/intellij-common/0.0.1-SNAPSHOT/maven-metadata.xml'. Received status code 401
```

Comment (jeffmaury) [reply build.gradle:60]: You need to add it for SNAPSHOT versions and I don't think the whole JBoss Nexus is cloned on Maven Central

Comment (adietish) [reply build.gradle:60]: > You need to add it for SNAPSHOT versions 

it is in my `~/.m2/settings.xml` for the url above via 
```
 <releases>
              <enabled>true</enabled>
              <updatePolicy>never</updatePolicy>
           </releases>
           <snapshots>
             <enabled>true</enabled>
             <updatePolicy>never</updatePolicy>
           </snapshots>
```

**Comment by adietish** [comment]
the PR should be alright and the new intellij-common with the factory is now published to repository.jboss.org and visible in nexus (https://repository.jboss.org/nexus/#nexus-search;quick~intellij-common). 
Nevertheless this new intellij-common is not available yet on in https://repository.jboss.org/nexus/#nexus-search;quick~intellij-common. There seems to be some caching issue. Waiting for it to disappear.

**Comment by adietish** [comment]
Merging. Closing.

**Pull request merged**